File: reimi/backend/reimi_app/src/main/java/com/reimi/reimi_app/domain/model/weatherpersonality/WeatherPersonalityCode.java
package com.reimi.reimi_app.domain.model.weatherpersonality;

public enum WeatherPersonalityCode {

    SPOE,
    SPOR,
    SPIE,
    SPIR,

    SFOE,
    SFOR,
    SFIE,
    SFIR,

    NPOE,
    NPOR,
    NPIE,
    NPIR,

    NFOE,
    NFOR,
    NFIE,
    NFIR;

    public static WeatherPersonalityCode from(String code) {
        try {
            return WeatherPersonalityCode.valueOf(code);
        } catch (IllegalArgumentException | NullPointerException e) {
            throw new IllegalArgumentException("存在しないタイプコードです");
        }
    }
}

File: reimi/frontend/reimi_app/lib/domain/value_objects/weather_personality_code.dart
import 'package:json_annotation/json_annotation.dart';

enum WeatherPersonalityCode {
  @JsonValue('SPOE')
  spoe,
  @JsonValue('SPOR')
  spor,
  @JsonValue('SPIE')
  spie,
  @JsonValue('SPIR')
  spir,

  @JsonValue('SFOE')
  sfoe,
  @JsonValue('SFOR')
  sfor,
  @JsonValue('SFIE')
  sfie,
  @JsonValue('SFIR')
  sfir,

  @JsonValue('NPOE')
  npoe,
  @JsonValue('NPOR')
  npor,
  @JsonValue('NPIE')
  npie,
  @JsonValue('NPIR')
  npir,

  @JsonValue('NFOE')
  nfoe,
  @JsonValue('NFOR')
  nfor,
  @JsonValue('NFIE')
  nfie,
  @JsonValue('NFIR')
  nfir;

  String get displayCode {
    return name.toUpperCase();
  }
}
